packages/@aws-cdk/aws-eks/test/test.user-data.ts

```diff
@@ -155,12 +157,18 @@ function newFixtures(spot = false) {
   const app = new App();
   const stack = new Stack(app, 'my-stack', { env: { region: 'us-west-33' } });
   const vpc = new ec2.Vpc(stack, 'vpc');
+  const cluster = Cluster.fromClusterAttributes(stack, 'cluster', {
+    clusterName: 'my-cluster-name',
+    clusterCertificateAuthorityData: 'CADATA',
+    clusterEndpoint: 'https://mycluster:443',
```

Comment:
Let's add a test with an imported cluster that doesn't specify these properties, to make sure the exceptions are handled properly.

Reply:
> So...thats conceptually true, but notice that user-data.ts is not exported from index.ts, so its essentially a private API. And as far as our code goes, we only call it from connectAutoScalingGroupCapacity .. Which is only available on Cluster, not ICluster.

Ok .. so in that case I'm fine with assuming these properties are always defined, and I don't need to worry about catching/handling exceptions?

I think we went down this path _because_ we wanted to make it work with imported clusters (without raising exceptions) too.  I'm lost on what problem I'm trying to address here :P

Reply:
> I'm lost on what problem I'm trying to address here :P

Ha yeah I think we got ourselves a little tangled here :)

> Ok .. so in that case I'm fine with assuming these properties are always defined, and I don't need to worry about catching/handling exceptions?

I would say that in that case it's better to just have this function accept `Cluster`, which will always have these properties defined. 

> I think we went down this path because we were concerned about backwards compatibility for imported clusters (ie: without raising exceptions)

Ah so yeah I guess its my bad, it wasn't about backwards compatibility ,I just didn't notice at the time that we can simply have this method accept `Cluster` instead of an `ICluster` - so I was trying to properly support that use case, with no need really.

So bottom line, lets just switch it to use `Cluster` :)

Reply:
Done.  The major change was to test.user-data.ts fixture/cases to avoid using ICluster.